tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -156,9 +156,34 @@ namespace tcf {
         tcf::error::ThrowIf<tcf::error::ValueError>(payload_format != "json-rpc",
             "Unsupported payload format found in the input");
 
-        // Decrypt Encryption key
-        ByteArray encrypted_session_key_bytes = HexStringToBinary(encrypted_session_key);
-        session_key = enclaveData.decrypt_message(encrypted_session_key_bytes);
+        try {
+            // Decrypt Encryption key
+            ByteArray encrypted_session_key_bytes = \
+                HexStringToBinary(encrypted_session_key);
+            session_key = \
+                enclaveData.decrypt_message(encrypted_session_key_bytes);
+        } catch (tcf::error::KeyRefreshError& e) {
+            Log(TCF_LOG_ERROR, "error::DecryptMessage - %d - %s", \
+                e.error_code(), e.what());
+            tcf::error::ThrowIf<tcf::error::KeyRefreshError>(
+                true == true, "Failed to decrypt encrypted session key. " \
```

Comment:
Use throw tcf::error::KeyRefreshError(msg) instead of  tcf::error::ThrowIf(true==true)

Reply:
Done